packages/core/src/components/toast/toast2.tsx

```diff
@@ -43,12 +43,15 @@ export const Toast2 = React.forwardRef<HTMLDivElement, ToastProps>((props, ref)
     const startTimeout = React.useCallback(() => setIsTimeoutStarted(true), []);
     const clearTimeout = React.useCallback(() => setIsTimeoutStarted(false), []);
 
+    // Per docs: "Providing a value less than or equal to 0 will disable the timeout (this is discouraged)."
+    const isTimeoutEnabled = timeout != null && timeout > 0;
+
     // timeout is triggered & cancelled by updating `isTimeoutStarted` state
     useTimeout(
         () => {
             triggerDismiss(true);
         },
-        isTimeoutStarted && timeout !== undefined ? timeout : null,
+        isTimeoutStarted && isTimeoutEnabled ? timeout : null,
```

Comment:
This fix looks simple. The hard part was making sure it was the right fix, which I ended up convincing myself is the case.